File: src/java/src/main/java/org/elastos/hive/vault/database/CountOptions.java
package org.elastos.hive.vault.database;

import com.google.gson.annotations.SerializedName;

public class CountOptions {
	@SerializedName("skip")
	private Long skip;

	@SerializedName("limit")
	private Long limit;

	@SerializedName("maxTimeMS")
	private Long maxTimeMS;

	public CountOptions setSkip(Long skip) {
		this.skip = skip;
		return this;
	}

	public CountOptions setLimit(Long limit) {
		this.limit = limit;
		return this;
	}

	public CountOptions setMaxTimeMS(Long maxTimeMS) {
		this.maxTimeMS = maxTimeMS;
		return this;
	}

	public Long getSkip() {
		return skip;
	}

	public Long getLimit() {
		return limit;
	}

	public Long getMaxTimeMS() {
		return maxTimeMS;
	}
}

File: src/service/database/countoptions.ts
/**
 * The options for the counting operation.
 */
export class CountOptions {
	private skip: number;
	private limit: number;
	private maxTimeMS: number;

	setSkip(skip: number): CountOptions {
	    this.skip = skip;
	    return this;
    }

    setLimit(limit: number): CountOptions {
        this.limit = limit;
        return this;
    }

    setMaxTimeMS(milliseconds: number): CountOptions {
        this.maxTimeMS = milliseconds;
        return this;
    }

    getSkip(): number {
	    return this.skip;
    }

    getLimit(): number {
        return this.limit;
    }

    getMaxTimeMS(): number {
        return this.maxTimeMS;
    }
}
